src/plugins/intel_cpu/src/nodes/reference.cpp

```diff
@@ -108,22 +108,38 @@ bool Reference::needShapeInfer() const {
 
 ov::TensorVector Reference::prepareInputs() const {
     ov::TensorVector inputs;
-    for (size_t i = 0; i < inputShapes.size(); i++) {
+    for (size_t i = 0lu; i < inputShapes.size(); i++) {
         void *srcDataPtr = getParentEdgesAtPort(i)[0]->getMemory().getData();
         ov::Shape shape = ovCoreNode->get_input_partial_shape(i).rank().get_length() == 0 ?
                 ov::Shape{} : getParentEdgesAtPort(i)[0]->getMemory().getStaticDims();
-        inputs.push_back(ov::Tensor(ovCoreNode->get_input_element_type(i), shape, srcDataPtr));
+
+        if (std::any_of(shape.begin(), shape.end(), [](const size_t dim) { return dim == 0lu; } )) {
+            inputs.push_back(ov::Tensor(ovCoreNode->get_input_element_type(i), shape));
+        } else {
+            if (!srcDataPtr) {
+                THROW_CPU_NODE_ERR("has empty input data on port ", i);
+            }
```

Comment:
NIT. This is rather an assert construction. Should we introduce `CPU_NODE_ASSERT`?

Reply:
Done